### PR TITLE
Assignment4

### PR DESCRIPTION
**Pull request by MikePHaynes**

### CODE REVIEW

File: GenericsHomework/GenericsHomework/Node.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                Node<T> newNode = new(value);
+                newNode.Next = Next;
+                Next = newNode;
+            }
+        }
+
+        public void Clear()
+        {
+            Next = this;
```
Comment (whitneybolar) [inline GenericsHomework/GenericsHomework/Node.cs:36]: In my code, I ended up not using <T>. I was not able to clear all of the extra nodes with just this.Next = this; I needed a while loop: while (i.Size != 1) // there should only be one left every time anyway.

File: GenericsHomework/GenericsHomework/Node.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
+﻿namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        public T Value { get; private set; }
```
Comment (whitneybolar) [inline GenericsHomework/GenericsHomework/Node.cs:5]: Is it mandatory to have a private set for Value and Next, or was that a personal choice? How come you choose that?

File: GenericsHomework/GenericsHomework.Tests/NodeTests.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void Append_ValueAlreadyExists_ThrowsException()
+        {
+            Node<string> node = new("Value");
+            node.Append("Value");
+        }
+
+        [TestMethod]
+        public void Append_ValuesDoNotAlreadyExist_Success()
```
[the remark this answers is not in the document]
Comment (whitneybolar) [reply GenericsHomework/GenericsHomework.Tests/NodeTests.cs:18]: I added a GetSize() property to help test if the number of nodes is correct like for Append or Clear(). That can be an option for testing.

[the remark this answers is not in the document]
Comment (whitneybolar) [reply GenericsHomework/GenericsHomework.Tests/NodeTests.cs:18]: You can test the constructor of Node by passing in a non null value to see if it comes out non null

File: GenericsHomework/GenericsHomework/Node.cs (hunk 2 above)
Comment (MikePHaynes) [reply GenericsHomework/GenericsHomework/Node.cs:5]: It was a personal choice. My reasoning for it was I didn't like the idea of it being possible to change the value of the nodes just to get unit tests to pass.

**Comment by MikePHaynes** [comment]
@whitneybolar Lots of good feedback here, thank you!

File: GenericsHomework/GenericsHomework.Tests/NodeTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikePHaynes) [reply GenericsHomework/GenericsHomework.Tests/NodeTests.cs:18]: Added ToString() test and constructor value test.

**Pull request closed without merging**